src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -179,6 +181,32 @@ export function registerActiveInstanceAction(
 	});
 }
 
+/**
+ * A wrapper around {@link registerTerminalAction} that ensures an active terminal
+ * exists and provides it to the run function.
+ *
+ * This includes detached xterm terminals that are not managed by an {@link ITerminalInstance}.
+ */
+export function registerActiveXtermAction(
+	options: IAction2Options & { run: (activeTerminal: XtermTerminal, accessor: ServicesAccessor, instance?: ITerminalInstance, args?: unknown) => void | Promise<unknown> }
+): IDisposable {
+	const originalRun = options.run;
+	return registerTerminalAction({
+		...options,
+		run: (c, accessor, args) => {
+			const activeDetached = Iterable.find(c.service.detachedInstances, d => d.isFocused);
+			if (activeDetached) {
+				return originalRun(activeDetached as XtermTerminal, accessor, undefined, args);
```

Comment:
We can remove this now that write/resize is on the interface?

Reply:
There's still a few other methods (getSelection, clearSelection, focus, selectAll, copySelection.) Happy to add these to the interface if you think that's preferable.

Reply:
Detached interface could extend IXtermTerminal? Those can be added to IXtermTerminal if they're not already as they're pretty safe